Composer/packages/client/src/pages/knowledge-base/QnAPage.tsx

```diff
@@ -26,13 +26,13 @@ const QnAPage: React.FC<RouteComponentProps<{
   projectId: string;
   skillId: string;
 }>> = (props) => {
-  const { dialogId = '', projectId = '', skillId } = props;
-
+  const { dialogId = '', projectId = '', skillId = '' } = props;
+  const currentBotProjectId = skillId ?? projectId;
```

Comment:
when skillId is undefined, it gets the empty string '' default and then your currentBotProjectId becomes empty string as well, is that what you want?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
I think skillID needs to be instantiated to null

Reply:
Careful here. The coalescing operator does _not_ treat `''` as a null value, so if there were no skillId given, the current it would end up with an empty string. That's why I had it without a default (because `undefined` would let the projectId pass through). @hatpick's comment about the projectId possibly also being empty is never going to happen, though; the QnAPage component gets values from the Router, and projectId is going to be read from the URL. The only reason it has a default value here is because TypeScript isn't aware of that fact.

Reply:
skillId is now left without instantiation